### PR TITLE
NIFI-9437 Flowfile Expiration cannot exceed 24 days

### PR DESCRIPTION
**Pull request by timeabarna**
# Summary

[NIFI-9537](https://issues.apache.org/jira/browse/NIFI-9437) enables setting flowfile expiration longer than 24 days.

# Tracking

Please complete the following tracking steps prior to pull request creation.

### Issue Tracking

- [x] [Apache NiFi Jira](https://issues.apache.org/jira/browse/NIFI) issue created

### Pull Request Tracking

- [x] Pull Request title starts with Apache NiFi Jira issue number, such as `NIFI-00000`
- [x] Pull Request commit message starts with Apache NiFi Jira issue number, as such `NIFI-00000`

### Pull Request Formatting

- [x] Pull Request based on current revision of the `main` branch
- [x] Pull Request refers to a feature branch with one commit containing changes

# Verification

Please indicate the verification steps performed prior to pull request creation.

### Build

- [ ] Build completed using `mvn clean install -P contrib-check`
  - [x] JDK 8
  - [ ] JDK 11
  - [ ] JDK 17

### Licensing

- [ ] New dependencies are compatible with the [Apache License 2.0](https://apache.org/licenses/LICENSE-2.0) according to the [License Policy](https://www.apache.org/legal/resolved.html)
- [ ] New dependencies are documented in applicable `LICENSE` and `NOTICE` files

### Documentation

- [ ] Documentation formatting appears as expected in rendered files

### CODE REVIEW

**Comment by thenatog** [comment]
Will review

File: nifi-stateless/nifi-stateless-bundle/nifi-stateless-engine/src/main/java/org/apache/nifi/stateless/queue/StatelessFlowFileQueue.java [hunk 1]
```diff
@@ -283,8 +283,8 @@ public String getFlowFileExpiration() {
     }
 
     @Override
-    public int getFlowFileExpiration(final TimeUnit timeUnit) {
-        return (int) timeUnit.convert(expirationMillis, TimeUnit.MILLISECONDS);
+    public long getFlowFileExpiration(final TimeUnit timeUnit) {
+        return timeUnit.convert(expirationMillis, TimeUnit.MILLISECONDS);
```
Comment (thenatog) [inline nifi-stateless/nifi-stateless-bundle/nifi-stateless-engine/src/main/java/org/apache/nifi/stateless/queue/StatelessFlowFileQueue.java:287]: I believe a change also needs to be made to   _public void setFlowFileExpiration(final String flowExpirationPeriod)_, as the value on line 292 is still being cast to int.

**Comment by thenatog** [comment]
I tested this out with a unit test in StatelessFlowFileQueue and found that the limit was still 24 days, which seems to be a result of the setFlowFileExpiration() method still casting to int. I also tested the function at runtime still works correctly for small times like 10 seconds. I did not verify that the expiration now works for > 24 days.

**Comment by timeabarna** [comment]
Hello @thenatog,

Thank you very much for your review. I hesitated whether flowfile expiration should be corrected as even 24 days seemed too long for stateless NiFI however you are right, let's be consistent. I've modified the code accordingly.

**Comment by thenatog** [comment]
+1 looks good, will merge

**Pull request closed without merging**